### PR TITLE
Rename ANDROID_ vars to CMAKE_ vars.

### PR DESCRIPTION
**Pull request by xsacha**
Also update Hunter version.

### CODE REVIEW

**Comment by xsacha** [comment]
Note: CMake 3.7+ Android doesn't seem to work (and never has).
Gives the following error:
```
                 from /home/sacha/dev/opencv/_builds/android-ndk-r14-api-21-arm64-v8a-Release/modules/hal/precomp.hpp:43:
/home/sacha/Android/Sdk/ndk-bundle/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/lib/gcc/aarch64-linux-android/4.9.x/include-fixed/limits.h:168:61: error: no include path in which to search for limits.h
 #include_next <limits.h>  /* recurse down to the real one */
```
```
                 from /home/sacha/dev/opencv/_builds/android-ndk-r14-api-21-arm64-v8a-Release/modules/hal/precomp.hpp:43:
make[2]: Leaving directory '/home/sacha/dev/opencv/_builds/android-ndk-r14-api-21-arm64-v8a-Release'
/home/sacha/Android/Sdk/ndk-bundle/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/lib/gcc/aarch64-linux-android/4.9.x/include/stdint.h:9:26: fatal error: stdint.h: No such file or directory
 # include_next <stdint.h>
```

Tested with Android NDK r14

File: 3rdparty/ippicv/downloader.cmake [hunk 1]
```diff
@@ -13,7 +13,7 @@ function(_icv_downloader)
     set(OPENCV_ICV_PLATFORM "macosx")
     set(OPENCV_ICV_PACKAGE_SUBDIR "/ippicv_osx")
   elseif(UNIX)
-    if(ANDROID AND (NOT ANDROID_ABI STREQUAL x86))
+    if(ANDROID AND (NOT CMAKE_ANDROID_ARCH STREQUAL x86))
```
Comment (ruslo) [inline 3rdparty/ippicv/downloader.cmake:16]: Should be `CMAKE_ANDROID_ARCH_ABI`, see:
* http://polly.readthedocs.io/en/latest/toolchains/android/old.html

File: cmake/OpenCVFindLibsGrfmt.cmake [hunk 2]
```diff
@@ -9,8 +9,8 @@ else()
   hunter_add_package(ZLIB)
   find_package(ZLIB "${MIN_VER_ZLIB}")
   if(ZLIB_FOUND AND ANDROID)
-    if(ZLIB_LIBRARIES STREQUAL "${ANDROID_SYSROOT}/usr/lib/libz.so" OR
-        ZLIB_LIBRARIES STREQUAL "${ANDROID_SYSROOT}/usr/lib64/libz.so")
+    if(ZLIB_LIBRARIES STREQUAL "${CMAKE_SYSROOT}/usr/lib/libz.so" OR
```
Comment (ruslo) [inline cmake/OpenCVFindLibsGrfmt.cmake:12]: Actually here we should always use ZLIB from Hunter. Should be protected with `NOT HUNTER_ENABLED`.

File: cmake/OpenCVGenConfig.cmake [hunk 3]
```diff
@@ -19,10 +19,10 @@ else()
   set(OpenCV_CUDA_VERSION ${CUDA_VERSION_STRING})
 endif()
 
-if(NOT ANDROID_NATIVE_API_LEVEL)
+if(ANDROID AND NOT CMAKE_SYSTEM_VERSION)
```
Comment (ruslo) [inline cmake/OpenCVGenConfig.cmake:22]: There was no `ANDROID` in old code.

**Review by ruslo — CHANGES_REQUESTED** [review]

**Comment by ruslo** [comment]
> Note: CMake 3.7+ Android doesn't seem to work (and never has).

`android-ndk-r10e-api-19-armeabi-v7a-neon` should work. See `3.1.0` build for example:
* https://travis-ci.org/ingenue/hunter/jobs/186525191

Also `3.0.0-*` should work too.

File: cmake/OpenCVGenConfig.cmake (hunk 3 above)
Comment (xsacha) [reply cmake/OpenCVGenConfig.cmake:22]: ANDROID was used on line 72. I assumed it is valid.

File: cmake/OpenCVFindLibsGrfmt.cmake (hunk 2 above)
Comment (xsacha) [reply cmake/OpenCVFindLibsGrfmt.cmake:12]: OK. I was just leaving previous functionality.

File: 3rdparty/ippicv/downloader.cmake (hunk 1 above)
Comment (xsacha) [reply 3rdparty/ippicv/downloader.cmake:16]: I agree for a search/replace but this condition is simply to check the architecture, not arch+abi.

**Comment by xsacha** [comment]
Interesting, I can't get 3.0.0-p9 or 3.1.0 working with latest hunter+polly. Gives error from first comment. Seems to be attempting to use local headers. Tried ndk r13/r14 on four different machines.

Last time it worked for me was hunter v16 (https://github.com/ruslo/hunter/releases/tag/v0.16.36)
All my projects are stuck on this version until this OpenCV issue is resolved.

**Comment by ruslo** [comment]
>  Tried ndk r13/r14 on four different machines

On Travis CI and my local machine it's `r10e`.

**Review by ruslo — APPROVED** [review]

**Comment by ruslo** [comment]
Released: https://github.com/hunter-packages/opencv/releases/tag/v3.0.0-p10

**Pull request merged**